src/components/drawing-widget/drawing-widget.component.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useEffect, useCallback } from "react";
+import ReactImageAnnotate, { Annotation } from "react-image-annotate";
+import { Add, Crop } from "@carbon/react/icons";
+import { useTranslation } from "react-i18next";
+import { CardHeader } from "@openmrs/esm-patient-common-lib";
+import { Button } from "@carbon/react";
+
+
+interface RegionData {
+  type: string;
+  x: number;
+  y: number;
+}
+
+export interface ImageData {
+  src: string;
+  name: string;
+  regions: RegionData[];
+}
+
+interface DrawingWidgetProps {
+  selectedImage: string;
+  taskDescription: string;
+  imagesData: ImageData[];
+  regionClsList?: string[];
+  enabledTools?: string[];
+  onExit: (annotations: ImageData[]) => void;
+}
+
+const DrawingWidget: React.FC<DrawingWidgetProps> = ({ onExit }) => {
+  const [selectedFile, setSelectedFile] = useState<File | null>(null);
+  const [selectedImage] = useState<string | null>(null);
+  const [annotations, setAnnotations] = useState<ImageData[]>([]);
+  const [, setLoading] = useState<boolean>(true);
+  const [activeImage, setActiveImage] = useState<ImageData | null>(null);
+  const { t } = useTranslation();
+ 
+
+  const CreatePointIcon = () => <Add />;
+  const CreateBoxIcon = () => <Crop />;
+
+  useEffect(() => {
+    const selectedImageURLParam = new URLSearchParams(
+      window.location.search
+    ).get("image-url");
```

Comment:
You use `useParams` in DrawPage, why can't you do the same here?

Reply:
Though actually I'd much rather you do all the url-param-getting in `DrawPage` and have this component take the image URL as a function parameter